### PR TITLE
Remove spaces in library name

### PR DESCRIPTION
**Pull request by FoamyGuy**
With these changes the user can choose a library name with spaces such as `LED Animation Core` and the resulting project files will get generated with underscores or dashes swapped in as needed.

I tested by running locally using the above library name. The resulting project folder now comes out as `Adafruit_CircuitPython_LED_Animation_Core` the python file is now `adafruit_led_animation_core.py` 

Several other references inside of the python file, readme, and setup.py have been updated as well.

### CODE REVIEW

**Comment by askpatrickw** [comment]
@FoamyGuy  Is this done? If so, I'm happy to run some tests and review

**Comment by FoamyGuy** [comment]
Whoops, forgot to request review. @askpatrickw yep, to the best of my knowledge it is done.

**Review by kattni — APPROVED** [review]

**Comment by kattni** [comment]
Please test before merging!

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/README.rst [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 {% set full_repo_name = cookiecutter.github_user + "/" + repo_name -%}
 {%- set pypi_name = cookiecutter.library_name|lower|replace("_", "-") -%}
 Introduction
 ============
 
 {% if cookiecutter.sphinx_docs | lower in ["yes", "y"] %}
-.. image:: https://readthedocs.org/projects/{% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | lower | replace("_", "-")}}-{% endif %}circuitpython-{{ cookiecutter.library_name | lower }}/badge/?version=latest
+.. image:: https://readthedocs.org/projects/{% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | lower | replace("_", "-")}}-{% endif %}circuitpython-{{ cookiecutter.library_name | lower | replace(" ", "-") }}/badge/?version=latest
 {%- if cookiecutter.target_bundle == 'Adafruit' %}
     :target: https://circuitpython.readthedocs.io/projects/{{ cookiecutter.library_name | lower }}/en/latest/
```
Comment (askpatrickw) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/README.rst:17]: The target has spaces in it when I ran the cookiecutter.

```
.. image:: https://readthedocs.org/projects/adafruit-circuitpython-my-amazing-library/badge/?version=latest
    :target: https://circuitpython.readthedocs.io/projects/my amazing library/en/latest/
    :alt: Documentation Status
```

**Review by askpatrickw — CHANGES_REQUESTED** [review]
I created "My Amazing Library", then I searched for Amazing in the output and checked if the spaces, dashes, or underscore made sense.


- [x] README.RST: The Installing from PyPI section had some spaces in URLS. In the PyPi link and in the pop install examples
- [x] Setup.py: Each word is added a keyword... in my example that looked bad. Maybe its fine though.
- [ ] api.rst: The automodule was incorrect, it has spaces should have underscores
- [ ] conf.py: Spaces are not being added between the provided name and the work "Library". In my case since I also used Library. It looks like this: `Adafruit_CircuitPython_My Amazing LibraryLibrary`
- [ ] examples.rst: The literalinclude section has spaces, should have underscores (in that same caption as well)
- [ ] index.rst, the Download link has spaces and should have underscores

If you run this your self and test it the way I did, you should quickly see these.

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/README.rst (hunk 1 above)
Comment (FoamyGuy) [reply {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/README.rst:17]: Thanks for taking a look, I will make these changes tonight.

Comment (FoamyGuy) [reply {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/README.rst:17]: I think I've gotten all of those resolved with the latest commits.

**Review by askpatrickw — CHANGES_REQUESTED** [review]
Looks great @FoamyGuy . I only see one thing. 

In Readme.rst I there are spaces in the target URL for Read the Docs

```
.. image:: https://readthedocs.org/projects/adafruit-circuitpython-vanlife-fan-controller/badge/?version=latest
    :target: https://circuitpython.readthedocs.io/projects/vanlife fan controller/en/latest/
    :alt: Documentation Status
```

I tested the adafruit and community, with and without a prefix.

**Comment by kattni** [comment]
@FoamyGuy Can you take another look at this? I'm moving this repo from `master` to `main` and while it's pretty simple to move PRs, it's even easier not to have to.

**Review by kattni — APPROVED** [review]
Verified README now generates properly. Thanks so much for fixing this! I ran into it today, and this will help me out.

**Pull request merged**